cmd/state-remote-installer/main.go

```diff
@@ -0,0 +1,204 @@
+		fmt.Fprintln(os.Stderr, err.Error())
+		exitCode = 1
+		return
+	}
+
+	an = sync.New(cfg, nil)
+
+	// Set up prompter
+	prompter := prompt.New(true, an)
+
+	params := newParams()
+	cmd := captain.NewCommand(
+		"state-installer",
+		"",
+		"Installs or updates the State Tool",
+		primer.New(nil, out, nil, nil, nil, nil, cfg, nil, nil, an),
+		[]*captain.Flag{ // The naming of these flags is slightly inconsistent due to backwards compatibility requirements
+			{
+				Name:        "channel",
+				Description: "Defaults to 'release'.  Specify an alternative channel to install from (eg. beta)",
+				Value:       &params.branch,
+			},
+			{
+				Shorthand: "b", // backwards compatibility
+				Hidden:    true,
+				Value:     &params.branch,
+			},
+			{
+				Name:        "version",
+				Shorthand:   "v",
+				Description: "The version of the State Tool to install",
+				Value:       &params.version,
+			},
+			{
+				Name:        "force",
+				Shorthand:   "f",
+				Description: "Force the installation, overwriting any version of the State Tool already installed",
+				Value:       &params.force,
+			},
+		},
+		[]*captain.Argument{},
+		func(ccmd *captain.Command, args []string) error {
+			return execute(out, prompter, cfg, an, args, params)
+		},
+	)
+
+	err = cmd.Execute(os.Args[1:])
+	if err != nil {
+		if locale.IsInputError(err) {
+			logging.Error("Installer input error: " + errs.JoinMessage(err))
+		} else {
+			multilog.Critical("Installer error: " + errs.JoinMessage(err))
+		}
+
+		exitCode = errs.UnwrapExitCode(err)
+		out.Error(err.Error())
+		return
+	}
+}
+
+func execute(out output.Outputer, prompt prompt.Prompter, cfg *config.Instance, an analytics.Dispatcher, args []string, params *Params) error {
+	msg := locale.Tr("tos_disclaimer", constants.TermsOfServiceURLLatest)
+	msg += locale.Tr("tos_disclaimer_prompt", constants.TermsOfServiceURLLatest)
+	cont, err := prompt.Confirm(locale.Tr("install_remote_title"), msg, p.BoolP(true))
+	if err != nil {
+		return errs.Wrap(err, "Could not prompt for confirmation")
+	}
+
+	if !cont {
+		return locale.NewInputError("install_cancel", "Installation cancelled")
+	}
+
+	branch := params.branch
+	if branch == "" {
+		branch = constants.ReleaseBranch
+	}
+
+	// Fetch payload
+	checker := updater.NewDefaultChecker(cfg)
+	checker.InvocationSource = updater.InvocationSourceInstall // Installing from a remote source is only ever encountered via the install flow
+	checker.VerifyVersion = false
+	update, err := checker.CheckFor(branch, params.version)
+	if err != nil {
+		return errs.Wrap(err, "Could not retrieve install package information")
+	}
+	if update == nil {
+		return errs.New("No update information could be found.")
+	}
+
+	version := update.Version
+	if params.branch != "" {
+		version = fmt.Sprintf("%s (%s)", version, branch)
+	}
+
+	out.Fprint(os.Stdout, fmt.Sprintf("• Downloading State Tool version [NOTICE]%s[/RESET]... ", version))
```

Comment:
We have access to locale, so why not localize this and subsequent user-facing messages?